### PR TITLE
feat: new component for data models - DIDs, signatures, JSON-LD

### PR DESCRIPTION
**Pull request by Moopli**
component/models/:
- did: afgo's DID and DID Doc models.
- ld: JSON-LD support - data models, context stores, LD-Proofs, canonicalization, etc.
- signature: signing APIs used in DID Docs etc.
- util: utility data models needed for other model implementations, but not specific to a given model.

### CODE REVIEW

**Comment by Moopli** [comment]
Note: Since this PR introduces a new component that afgo depends on, it needs a followup that removes the go.mod replace directives (`replace github.com/hyperledger/aries-framework-go/component/models => ./component/`, etc) and updates imports of `component/models` to use this PR commit.

**Review by sandrask — APPROVED** [review]

**Review by rolsonquadras — APPROVED** [review]

File: component/models/ld/mock/mock_context_store.go [hunk 1]
```diff
@@ -13,14 +13,14 @@ import (
...
-	"github.com/hyperledger/aries-framework-go/pkg/store/ld"
+	"github.com/hyperledger/aries-framework-go/component/models/ld/context"
+	store "github.com/hyperledger/aries-framework-go/component/models/ld/store"
+	mockstorage "github.com/hyperledger/aries-framework-go/component/storageutil/mock/storage"
 	"github.com/hyperledger/aries-framework-go/spi/storage"
 )
 
 // MockContextStore is a mock JSON-LD context store.
-type MockContextStore struct {
+type MockContextStore struct { // nolint:golint
```
Comment (DRK3) [inline component/models/ld/mock/mock_context_store.go:23]: Reason for the `nolint`?

**Review by vkubiv — APPROVED** [review]

File: component/models/ld/processor/processor.go
Comment (DRK3) [inline component/models/ld/processor/processor.go:1]: The package name is still `jsonld` - it should be changed to `processor`

File: component/models/ld/documentloader/document_loader.go
Comment (DRK3) [inline component/models/ld/documentloader/document_loader.go:1]: Package name needs to be updated to avoid weirdness when importing

**Comment by DRK3** [comment]
Something else I'm wondering: do we really need individual packages for all the various pieces under `component/models/ld`? If they're all related, is there a reason to require a caller to import multiple packages? Will anyone need to import one of them but not the other? etc

File: component/models/ld/documentloader/document_loader.go
Comment (Moopli) [reply component/models/ld/documentloader/document_loader.go:1]: done!

File: component/models/ld/processor/processor.go
Comment (Moopli) [reply component/models/ld/processor/processor.go:1]: done!

File: component/models/ld/processor/processor_test.go [hunk 2]
```diff
@@ -14,9 +14,9 @@ import (
 
 	"github.com/stretchr/testify/require"
 
-	"github.com/hyperledger/aries-framework-go/pkg/doc/ldcontext"
-	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/jsonld"
-	"github.com/hyperledger/aries-framework-go/pkg/internal/ldtestutil"
+	ldcontext "github.com/hyperledger/aries-framework-go/component/models/ld/context"
+	"github.com/hyperledger/aries-framework-go/component/models/ld/ldtestutil"
```
[the remark this answers is not in the document]
Comment (Moopli) [reply component/models/ld/processor/processor_test.go]: done!

File: component/models/ld/mock/mock_context_store.go (hunk 1 above)
Comment (Moopli) [reply component/models/ld/mock/mock_context_store.go]: This is for the name stuttering, but I've now fixed the stuttering instead.

File: component/models/ld/store/context_store.go
[the remark this answers is not in the document]
Comment (Moopli) [reply component/models/ld/store/context_store.go:1]: fixed!

**Comment by Moopli** [comment]
> If they're all related, is there a reason to require a caller to import multiple packages? Will anyone need to import one of them but not the other?

Hmm so, `testutil` definitely needs to be separate, and usually only `context` is imported together with other `ld/` packages, so I'm not sure what it gains, while at least right now there's a clear separation of functionality.

**Review by DRK3 — APPROVED** [review]

**Pull request merged**